numpy/_core/src/multiarray/scalartypes.cpp

```diff
@@ -2103,72 +2328,321 @@ static PyObject *
+{
+    return gentype_generic_method(self, args, kwds, "sort");
+}
+
+static PyObject *
+gentype_argsort(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    return gentype_generic_method(self, args, kwds, "argsort");
+}
+
+static PyObject *
+gentype_round(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    return gentype_generic_method(self, args, kwds, "round");
+}
+
+static PyObject *
+gentype_argmax(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    return gentype_generic_method(self, args, kwds, "argmax");
+}
+
+static PyObject *
+gentype_argmin(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    return gentype_generic_method(self, args, kwds, "argmin");
+}
+
+static PyObject *
+gentype_max(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    return gentype_generic_method(self, args, kwds, "max");
+}
+
+static PyObject *
+gentype_min(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    return gentype_generic_method(self, args, kwds, "min");
+}
+
+static PyObject *
+gentype_any(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    return gentype_generic_method(self, args, kwds, "any");
+}
+
+static PyObject *
+gentype_all(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    return gentype_generic_method(self, args, kwds, "all");
+}
+
+static PyObject *
+gentype_astype(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    return gentype_generic_method(self, args, kwds, "astype");
+}
+
+static PyObject *
+gentype_resize(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    return gentype_generic_method(self, args, kwds, "resize");
+}
+
+static PyObject *
+gentype_reshape(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    return gentype_generic_method(self, args, kwds, "reshape");
+}
+
+static PyObject *
+gentype_choose(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    return gentype_generic_method(self, args, kwds, "choose");
+}
+
+static PyObject *
+gentype_tobytes(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    return gentype_generic_method(self, args, kwds, "tobytes");
+}
+
+static PyObject *
+gentype_copy(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    return gentype_generic_method(self, args, kwds, "copy");
+}
+
+static PyObject *
+gentype_searchsorted(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    return gentype_generic_method(self, args, kwds, "searchsorted");
+}
+
+static PyObject *
+gentype_view(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    return gentype_generic_method(self, args, kwds, "view");
+}
+
+static PyObject *
+gentype_flatten(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    return gentype_generic_method(self, args, kwds, "flatten");
+}
+
+static PyObject *
+gentype_ravel(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    return gentype_generic_method(self, args, kwds, "ravel");
+}
+
+static PyObject *
+gentype_squeeze(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    return gentype_generic_method(self, args, kwds, "squeeze");
+}
```

Comment:
given that it's trivial it's probably better to make this a macro though, it'll be a lot more concise.